components/tools/OmeroWeb/omeroweb/webgateway/marshal.py

```diff
@@ -95,15 +108,17 @@ def imageMarshal(image, key=None, request=None):
             'projectName': pr and pr.name or 'Multiple',
             'projectId': pr and pr.id or None,
             'projectDescription': pr and pr.description or '',
-            'datasetName': ds and ds.name or 'Multiple',
-            'datasetId': ds and ds.id or None,
-            'datasetDescription': ds and ds.description or '',
-            'wellSampleId': wellsample and wellsample.id or '',
-            'wellId': well and well.id.val or '',
+            'datasetName': ds and ds['name'] or 'Multiple',
```

Comment:
could you change to `a if test else b`, https://www.python.org/dev/peps/pep-0308/